src/listr/tasks/startNodeTaskFactory.js

```diff
@@ -55,9 +52,76 @@ function startNodeTaskFactory(dockerCompose) {
           }
         },
       },
+      {
+        title: 'Download updated services',
+        task: async (ctx, task) => {
+          // async function waitStream() {
+          //   // eslint-disable-next-line no-new
+          //   new Promise((resolve, reject) => {
+          //     const pullCommand = dockerCompose.pull(config.toEnvs());
+          //     // eslint-disable-next-line no-param-reassign
+          //     task.output = pullCommand.out;
+          //     pullCommand.then(resolve)
+          //       .catch(() => reject(new Error('Updating services failed')));
+          //   });
+          // }
+          // await waitStream();
+
+          // Doesn't wait
+
+          // ---------------------------------------------------------------------
+
+          // await new Promise((resolve, reject) => {
+          //   const readable = Readable.from(dockerCompose.pull(config.toEnvs()));
+          //   readable.on('data', (chunk) => {
+          //     // eslint-disable-next-line no-param-reassign
+          //     task.output = chunk;
+          //   });
+          //   readable.on('end', resolve);
+          //   readable.on('error', (error) => {
+          //     throw new Error(error);
+          //   });
+          // });
+
+          // The "iterable" argument must be an instance of Iterable. Received an instance of Promise
+
+          // ---------------------------------------------------------------------
+
+          // let streamy = '';
+          // const theObserver = new Observable((observer) => {
+          //   streamy = from(dockerCompose.pull(config.toEnvs()));
```

Comment:
You forgot `await` as well. The pull method responds with a promise which in turn resolves with the stream.